src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -95,7 +74,9 @@ public String call() throws Exception {
                 CriteriaQuery<Long> query = criteriaBuilder.createQuery(Long.class);
                 Root<DeviceEntity> r = query.from(DeviceEntity.class);
                 query.select(r.get("id").as(Long.class)).where(criteriaBuilder.equal(r.get("name"), name));
+
                 List<Long> result = IsmDeviceApi.this.em.createQuery(query).getResultList();
```

Comment:
you should find the device under the vs. Names can be duplicated across VS'es.

In the current implementation, result list could possibly contain more than 1 and you would just return the first result.

Same comment applicables for other api's

Reply:
There is no VS entity on the plugin db. Maybe this can be done by adding the vs id as a new column on this table and query by id and name. @arvindn05 thoughts?